src/commands/tag-image.yml

```
@@ -0,0 +1,25 @@
+description: Add a tag to an existing published image
+
+parameters:
+  repo:
+    type: string
+    description: Name of an Amazon ECR repository
+
+  source-tag:
+    type: string
+    description: An existing Docker image tag
+
+  target-tag:
+    type: string
+    description: A comma-separated string containing docker image tags (default = latest)
+
+steps:
+  - run:
+      name: Add <<parameters.target-tag>> tag to <<parameters.repo>>:<<parameters.source-tag>>
+      command: |
+        # pull the image manifest from ECR
+        MANIFEST=$(aws ecr batch-get-image --repository-name <<parameters.repo>> --image-ids imageTag=<<parameters.source-tag>> --query 'images[].imageManifest' --output text)
+        IFS="," read -ra ECR_TAGS \<<< "<< parameters.target-tag >>"
+        for tag in "${ECR_TAGS[@]}"; do
+          aws ecr put-image --repository-name <<parameters.repo>> --image-tag ${tag} --image-manifest "$MANIFEST"
```

```suggestion
        MANIFEST=$(aws ecr batch-get-image --repository-name "<<parameters.repo>>" --image-ids "imageTag=<<parameters.source-tag>>" --query 'images[].imageManifest' --output text)
        IFS="," read -ra ECR_TAGS \<<< "<< parameters.target-tag >>"
        for tag in "${ECR_TAGS[@]}"; do
          aws ecr put-image --repository-name "<<parameters.repo>>" --image-tag "${tag}" --image-manifest "$MANIFEST"
```

I would suggest the interpolated values here be wrapped in quotes to ensure correct parsing by the shell as a single argument.